### PR TITLE
[LAUNCH] add distributed launch check tools

### PR DESCRIPTION
**Pull request by kuizhiqing**
### PR types

New features

### PR changes

Others

### Describe

Test your distributed configuration status by running
```
python -m paddle.distributed.launch test
```

### CODE REVIEW

File: python/paddle/distributed/launch/plugins/__init__.py [hunk 1]
```diff
@@ -60,4 +61,16 @@ def rewrite_host_ip(ctx):
         ctx.node.ip = ctx.args.host
 
 
-enabled_plugins = [collective_compatible, rewrite_host_ip, process_args]
+def test_mode(ctx):
+    if ctx.args.training_script == 'test':
```
Comment (sneaxiy) [inline python/paddle/distributed/launch/plugins/__init__.py:65]: Recommend to rename to `run_check`. You can keep the name  in this PR. The final name should be discussed offline in the meeting.

File: python/paddle/distributed/launch/controllers/collective.py [hunk 2]
```diff
@@ -97,10 +97,14 @@ def build_pod(self):
                 "PADDLE_TRAINERS_NUM": "{}".format(global_size),
                 "PADDLE_RANK_IN_NODE": str(i),
             }
-            if self.pod.replicas == 1:
-                e.update({selected_dev_key: ",".join(selected_dev_list)})
+            if len(selected_dev_list) > 0:
```
Comment (sneaxiy) [inline python/paddle/distributed/launch/controllers/collective.py:100]: Why change this file?

File: python/paddle/distributed/launch/context/__init__.py [hunk 3]
```diff
@@ -101,6 +101,7 @@ def continous_log(self) -> bool:
             return False
 
     def set_env_in_args(self):
+        # this logic may not propre to replace args with env, but ...
```
Comment (sneaxiy) [inline python/paddle/distributed/launch/context/__init__.py:104]: Why change this?

File: python/paddle/distributed/launch/controllers/collective.py (hunk 2 above)
Comment (kuizhiqing) [reply python/paddle/distributed/launch/controllers/collective.py:100]: Default to cpu/gloo mode if gpu is not used.

File: python/paddle/distributed/launch/context/__init__.py (hunk 3 above)
Comment (kuizhiqing) [reply python/paddle/distributed/launch/context/__init__.py:104]: bug fix

**Review by sneaxiy — APPROVED** [review]
LGTM. Should change `test` to `run_check` in the next PR.

**Pull request merged**